fix(podman_image): skip empty volume items

bmenant opened the pull request:
Skip empty volume items with `podman image build`:

```yaml
- containers.podman.podman_image:
    name: myimage
    path: /tmp/myimage
    build:
      volume:
        - "{{ '/tmp/whatever:/mnt/whatever' if false }}" # results in "" (empty string), which leads to podman error
    state: build
```

This PR adds a check in `podman_image` module, just like it is already done over here: https://github.com/containers/ansible-podman-collections/blob/master/plugins/module_utils/podman/podman_container_lib.py#L859-L863

sshnaidm commented:
Well, the condition is legit, but putting an empty string there is definitely wrong. If you want to exclude specific volumes it's better to prepare "volumes" variable before.

bmenant commented:
Well, I understand it’s wrong, but from an Ansible playbook point of view, would it hurt much if it was a little loose? It sometimes helps to write tasks more naturally IMHO.
At least, it should be consistent with `podman_container` module:

```yaml
---
- name: podman volume opts playbook
  hosts: localhost
  connection: local
  gather_facts: false

  vars:
    some_condition: false                                               # container | image (build)
    volopts:                                                            # ----------|--------------
      - "{{ [ 'host_dir:container_dir' ] if some_condition else [] }}"  # pass      | pass
      - "{{ [ 'host_dir:container_dir' ] if some_condition }}"          # pass      | fail
      -                                                                 #           |
        - "{{ 'host_dir:container_dir' if some_condition }}"            # pass      | fail 

  tasks:
    - name: podman_container volume
      loop: "{{ volopts }}"
      containers.podman.podman_container:
        volume: "{{ item }}"
        image: "quay.io/podman/hello"
        name: test-podman-container-volume
        recreate: true
        state: created 
    - name: podman_image volume
      loop: "{{ volopts }}"
      containers.podman.podman_image:
        build:
          container_file: "FROM scratch"
          volume: "{{ item }}"
        force: true
        name: test-podman-image-volume
        path: "{{ playbook_dir }}"
        state: build

```

```
PLAY [podman volume opts] ******************************************************************************************************************************************************************************************************

TASK [podman_container volume] *************************************************************************************************************************************************************************************************
changed: [localhost] => (item=[])
changed: [localhost] => (item=)
changed: [localhost] => (item=[''])

TASK [podman_image volume] *****************************************************************************************************************************************************************************************************
changed: [localhost] => (item=[])
failed: [localhost] (item=) => {"ansible_loop_var": "item", "changed": false, "item": "", "msg": "Failed to build image test-podman-image-volume:latest:  Error: creating build executor: incorrect volume format \"\", should be host-dir:ctr-dir[:option]\n"}
failed: [localhost] (item=['']) => {"ansible_loop_var": "item", "changed": false, "item": [""], "msg": "Failed to build image test-podman-image-volume:latest:  Error: creating build executor: incorrect volume format \"\", should be host-dir:ctr-dir[:option]\n"}

PLAY RECAP *********************************************************************************************************************************************************************************************************************
localhost                  : ok=1    changed=1    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0 
```

sshnaidm commented:
In your case it's possible either to select non-empty in `volopts`:
```yaml
-   name: podman_container volume
    loop: "{{ volopts | select('length') | list }}"
    containers.podman.podman_container:
      volume: "{{ item }}"
      image: "quay.io/podman/hello"
      name: test-podman-container-volume
      recreate: true
      state: created 
```

or to add `when` condition:

```yaml
-   name: podman_container volume
    loop: "{{ volopts }}"
    when: item | length > 0
    containers.podman.podman_container:
      volume: "{{ item }}"
      image: "quay.io/podman/hello"
      name: test-podman-container-volume
      recreate: true
      state: created
```

I'll merge it, but there are multiple ways to do it right instead.
